cmd/snap-confine/snap-confine.c

```diff
@@ -37,50 +37,48 @@
 #endif				// ifdef HAVE_SECCOMP
 #include "udev-support.h"
 #include "user-support.h"
+#include "snap-confine-args.h"
 
 int main(int argc, char **argv)
 {
-	if (argc == 2 && strcmp(argv[1], "--version") == 0) {
+	// Use our super-defensive parser to figure out what we've been asked to do.
+	struct sc_error *err = NULL;
+	struct sc_args *args __attribute__ ((cleanup(sc_cleanup_args))) = NULL;
+	args = sc_nonfatal_parse_args(&argc, &argv, &err);
+
+	if (sc_error_match(err, SC_ARGS_DOMAIN, SC_ARGS_ERR_USAGE)) {
+		sc_error_free(err);
+		// Let's print our nice usage string and exit.
+		fprintf(stderr, "Usage: %s <security-tag> <executable>",
+			argv[0]);
+		return 1;
+	}
+	// We don't want to handle any other errors, just die if we see one.
+	sc_die_on_error(err);
```

Comment:
This should be something like this instead:
```
err = sc_parse_args(&argc, &argv, &args);
sc_die_on_error(err);
```
The usage error may be an actual error inside the respective logic instead of just a code:
```
error *usage_error = errorf("Usage: snap-confine <security tag> <excutable>")
```